en/xml/fo/topic.graphic-pageheader.xml

```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+    License CC BY-NC-SA 3.0
+
+   This work is licensed under the
+   "Namensnennung – Keine kommerzielle Nutzung – Weitergabe unter
+    gleichen Bedingungen 3.0 Deutschland (CC BY-NC-SA 3.0)"
+   http://creativecommons.org/licenses/by-nc-sa/3.0/de/deed.de
+
+   Read the English translation here:
+
+   "Attribution-NonCommercial-ShareAlike 3.0 Unported (CC BY-NC-SA 3.0)"
+   http://creativecommons.org/licenses/by-nc-sa/3.0/
+
+-->
+<?xml-model href="file:../5.1/dbref.rnc" type="application/relax-ng-compact-syntax"?>
+<section xml:id="dbc.fo.graphic-pageheader" remap="topic" userlevel="easy"
+   xmlns="http://docbook.org/ns/docbook"
+   xmlns:xi="http://www.w3.org/2001/XInclude"
+   xmlns:xlink="http://www.w3.org/1999/xlink">
+   <title>Inserting Graphics to your Page Headers or Footers</title>
+   <info>
+       <author>
+           <personname>
+               <firstname>Peter</firstname>
+               <surname>Schmelzer</surname>
+           </personname>
+       </author>
+       <keywordset>
+         <keyword>page header</keyword>
+         <keyword>block</keyword>
+         <keyword>format</keyword>
+         <keyword>page</keyword>
+      </keywordset>
+      <subjectset>
+         <subject>
+            <subjectterm>graphic</subjectterm>
+         </subject>
+      </subjectset>
+   </info>
+   
+   <section role="problem">
+      <title>Problem</title>
+      <para>You need to know how to insert a graphic in your page header or footer.</para>
+   </section>
+
+   <section role="solution">
+      <title>Solution</title>
+      <para>First of all, you need to create a customization layer in your 
+         customization stylesheet containing the following code:</para>
+      <programlisting>&lt;!-- Path and name of your graphic file-->
+&lt;xsl:param name="header.image.filename">{$img.src.path}your-graphic.svg&lt;/xsl:param>
```

Comment:
I'm not sure what you mean. Do you mean this:
    <xsl:param name="header.image.filename">your-graphic.svg>/xsl:param>
    <xsl:param name="header.image.filename" select="concat($img.src.path, 'your-graphic.svg')"/>

Reply:
My idea was that `header.image.filename` contains only the file name and nothing else:

```xml
<xsl:param name="header.image.filename">your-graphic.svg</xsl:param>
```

The complete path would be built by concatenating it with `$img.src.path`. Does that make any sense? :wink:

Reply:
Ok. That makes sense. Thanks!
The complete solution looks like this?
    <xsl:param name="img.src.path">./Images/</xsl:param>
    <xsl:param name="header.image.filename">your-graphic.svg</xsl:param>
    xsl:param name="header.image.filename" select="concat($img.src.path, 'your-graphic.svg')"/>
I‘m not sure if this is correct, because 'your-graphic.svg' after concatenation is incorrect.

Reply:
Almost. :wink:  These lines are correct:

```xml
<xsl:param name="img.src.path">./Images/</xsl:param>
<xsl:param name="header.image.filename">your-graphic.svg</xsl:param>
```

A little bit deeper you use the parameter `header.image.filename` in this code:

```xml

&lt;xsl:call-template name="fo-external-image"><co xml:id="co.topic.graphic-pageheader.template"/>
     &lt;xsl:with-param name="filename" select="concat($img.src.path, $header.image.filename)"/><co xml:id="co.topic.graphic-pageheader.file"/>>
&lt;/xsl:call-template>
```
Keep in mind the following changes in comparison to your code:

* `xsl:param` inside `xsl:call-template` has to be `xsl:with-param`.
* the parameter for the template `fo-external-image` has to be `filename`, not `header.image.filename`.
* you pass the complete path in the `select` attribute of `xsl:with-param`.

Reply:
Done.